Makefile

```diff
@@ -1,28 +1,54 @@
 COMMIT?=$(shell git describe HEAD)$(shell git diff --quiet || echo '+dirty')
 
 # Use linker flags to provide commit info
-LDFLAGS=-ldflags "-X=github.com/foundriesio/fioctl/subcommands/version.Commit=$(COMMIT)"
+VERSION_LDFLAGS=-X=github.com/foundriesio/fioctl/subcommands/version.Commit=$(COMMIT)
+BUILD_STATIC=1
+PREFIX=bin/fioctl
 
 linter:=$(shell which golangci-lint 2>/dev/null || echo $(HOME)/go/bin/golangci-lint)
+builder:=$(shell which xgo 2>/dev/null || echo $(HOME)/go/bin/xgo)
+xgopkg::=src.techknowlogick.com/xgo@v1.7.0+1.19.5
+xgoimg::=ghcr.io/crazy-max/xgo:1.21.0
 
+# Platforms listed in build and build-hsm targets are fully supported and tested.
+# See the README for a full list of platforms tentatively supported by our XGo cross-compiler toolchain.
+
+# Statically linked binary without HSM support on platforms that support static binaries.
+# On platforms not supporting static binaries (e.g. Darwin) this builds dynamic library without HSM support.
 build: fioctl-linux-amd64 fioctl-linux-arm64 fioctl-windows-amd64 fioctl-darwin-amd64 fioctl-darwin-arm64
 	@true
 
-fioctl-static:
-	CGO_ENABLED=0 go build -a -ldflags '-w -extldflags "-static"' -o ./bin/fioctl-static ./main.go
-
-fioctl-linux-amd64:
-fioctl-linux-arm64:
-fioctl-linux-armv7:
-fioctl-windows-amd64:
-fioctl-darwin-amd64:
-fioctl-darwin-arm64:
-fioctl-%:
-	CGO_ENABLED=0 \
-	GOOS=$(shell echo $* | cut -f1 -d\- ) \
-	GOARCH=$(shell echo $* | cut -f2 -d\-) \
-		go build $(LDFLAGS) -o bin/$@ main.go
-	@if [ "$@" = "fioctl-windows-amd64" ]; then mv bin/$@ bin/$@.exe; fi
+# Dynamically linked binary with HSM support.
+# There is no way to build static binary with HSM support due to the PKCS11 design choice to use dlopen.
+# For Darwin and Windows we link against the lowest possible library version for the greatest compatibility.
+# For Linux we link against GNU Libc 6; so it must be installed on a target system.
+build-hsm: fioctl-hsm-darwin-amd64 fioctl-hsm-darwin-arm64 fioctl-hsm-windows-amd64 fioctl-hsm-libc6-linux-amd64
+	@true
+
+# Below 3 targets allow you to build the local binary using your host toolchain
+fioctl-local-dynlink:
+	go build -ldflags '-s -w' -o ./bin/fioctl-local-dynlink ./main.go
+
+fioctl-hsm-local-dynlink:
+	go build -tags=hsm -ldflags '-s -w' -o ./bin/fioctl-hsm-local-dynlink ./main.go
```

Comment:
Does this one also need `dynamic` added to the tags in order for it to trigger the hsm dynamic on windows? I base this on #270 .

Reply:
This is intended.

My initial idea was to prevent accidental build of a static binary with HSM support (which will certainly either fail or produce a corrupt binary).

But, this subject is more convoluted than I thought initially.
For example, there is no such thing as static binary on Darwin and few other platforms.
Which brings me to the idea that the whole "static/dynamic" thinking is kind of narrow-minded in our non-binary world.

So, it is easier to have just `hsm` tag to control if HSM support is being built.